### PR TITLE
Improve the dashboard buffer name and make it customizable

### PR DESCRIPTION
**Pull request by j-xella**
Dashboard creates a scratch buffer, named 'Scratch' by default. Not very intuitive.

This does not matter much if status/window/tab lines are hidden, the default behaviour. But may not look nice otherwise.

### CODE REVIEW

**Comment by glepnir** [comment]
sorry i don't get the point.

**Comment by j-xella** [comment]
In my environment, I prefer the tabline to be visible, and dashboard to appear every time I open a new tab. Config like this:

```lua
 db.setup( {                                            
   theme = 'hyper',                                     
   hide = {                                             
     tabline = false                                    
   },                                                   
   config = {                                           
     -- disable_move = true,                            
   }                                                    
 } )                                                    
                                                        
 vim.api.nvim_create_autocmd('TabNewEntered', {         
   desc = "Open dashboard-nvim on new tab",             
   callback = function() require('dashboard'):instance()
 })                                                     
```

This causes the dashboard on a new tab to look like this:

image

Note that the name of the new tab where the dashboard is opened is shown as `Scratch`. Dashboard creates a temporary buffer, and it is called "Scratch" by default. So wherever the current buffer name can be displayed, this word "Scratch" appears. This is not a big deal, just a cosmetic issue, but it does not look nice, IMO.

With my fix, it looks like this:

image

I have opened several new tabs in parallel to show how the fix behaves when multiple dashboards are opened

File: README.md [hunk 1]
```diff
@@ -53,6 +53,7 @@ theme = 'hyper' --  theme is doom and hyper default is hyper
 disable_move    --  defualt is false disable move keymap for hyper
 shortcut_type   --  shorcut type 'letter' or 'number'
 change_to_vcs_root -- default is false,for open file in hyper mru. it will change to the root of vcs
+buffer_name = 'Dashboard' -- the name of the dashboard's buffer. Visible depending on the nvim config
```
Comment (glepnir) [inline README.md:56]: does this option make sense ? can we use it as default.

File: lua/dashboard/init.lua [hunk 2]
```diff
@@ -192,6 +193,28 @@ function db:get_opts(callback)
   )
 end
 
+local function get_unique_buffer_name(opts)
```
Comment (glepnir) [inline lua/dashboard/init.lua:196]: maybe it can receive buffer number then just do a format .

File: README.md (hunk 1 above)
Comment (j-xella) [reply README.md:56]: User does not have to specify it explicitly, "Dashboard" is already the default value. But though this option user can change it, for example, to translate to another language.

Maybe I made it unclear in the help file?

File: lua/dashboard/init.lua (hunk 2 above)
Comment (j-xella) [reply lua/dashboard/init.lua:196]: This function actually does the format already, but not on a buffer number, but an integer from 1 to 9999, until it finds a unique buffer name. I don't think using buffer id is a good idea, it constantly increases, so we will soon end up with something like "Dashboard-77", which is confusing, IMO.

By default, the logic appends the integer at the end (like in the screenshot above), but if the name contains formatting sequences, like "%02d" for example, those will be used.

File: lua/dashboard/init.lua [hunk 3]
```diff
@@ -192,6 +193,28 @@ function db:get_opts(callback)
...
+      name2 = opts.buffer_name .. '-' .. i
+    end
+
+    if -1 == vim.fn.bufnr(name2) then
+      return name2
+    end
+  end
+
+  -- if we are here, then it is bad ... but chances of getting here are very low
+  error('Unable to find unique name for the dashboard buffer')
```
Comment (glepnir) [inline lua/dashboard/init.lua:215]: if not get it can we provide a default name not through an error ?

**Pull request closed without merging**